src/main/java/com/questhelper/helpers/quests/perilousmoon/PerilousMoon.java

```diff
@@ -0,0 +1,702 @@
+		goBuildEarthCamp.addStep(inEarthbound, buildEarthCamp);
+		goBuildEarthCamp.addStep(inPrison, leavePrison);
+		goBuildEarthCamp.addStep(inAntechamber, enterEarthbound);
+
+		ConditionalStep goBuildStreamCamp = goEnterNeypotzli.copy();
+		goBuildStreamCamp.addStep(inStreambound, buildStreamCamp);
+		goBuildStreamCamp.addStep(inEarthbound, leaveEarth);
+		goBuildStreamCamp.addStep(inPrison, leavePrison);
+		goBuildStreamCamp.addStep(inAntechamber, enterStreambound);
+
+		ConditionalStep buildCamps = goEnterNeypotzli.copy();
+		buildCamps.addStep(and(buildingSupplies, madePrisonCamp, madeEarthCamp), goBuildStreamCamp);
+		buildCamps.addStep(and(buildingSupplies, madePrisonCamp), goBuildEarthCamp);
+		buildCamps.addStep(buildingSupplies, goBuildPrisonCamp);
+		buildCamps.addStep(inAntechamber, takeBuildingSupplies);
+		steps.put(10, buildCamps);
+
+		ConditionalStep goReturnToJessamine = goEnterNeypotzli.copy();
+		goReturnToJessamine.addStep(inAntechamber, talkToJessamineForTalismans);
+		goReturnToJessamine.addStep(inStreambound, leaveStream);
+		goReturnToJessamine.addStep(inEarthbound, leaveEarth);
+		goReturnToJessamine.addStep(inPrison, leavePrison);
+		steps.put(12, goReturnToJessamine);
+		steps.put(13, goReturnToJessamine);
+		steps.put(14, goReturnToJessamine);
+
+		ConditionalStep goTalkToNahta = new ConditionalStep(this, enterCamTorum);
+		goTalkToNahta.addStep(inCamTorum, talkToNahta);
+		goTalkToNahta.addStep(inAntechamber, leaveNeypotzli);
+		steps.put(15, goTalkToNahta);
+
+		ConditionalStep goTalkToBlacksmith = new ConditionalStep(this, enterCamTorum);
+		goTalkToBlacksmith.addStep(inCamTorum, talkToBlacksmith);
+		goTalkToBlacksmith.addStep(inAntechamber, leaveNeypotzli);
+		steps.put(16, goTalkToBlacksmith);
+
+		ConditionalStep goFindEyatlalli = new ConditionalStep(this, enterCamTorum);
+		goFindEyatlalli.addStep(eyatlalliNearby, talkToEyatlalli);
+		goFindEyatlalli.addStep(inNeypotzli, useTalismans);
+		goFindEyatlalli.addStep(inCamTorum, enterNeypotzliAfterBlacksmith);
+		steps.put(17, goFindEyatlalli);
+		steps.put(18, goFindEyatlalli);
+
+		ConditionalStep goTalkToJessamineAfterEyatlalli = new ConditionalStep(this, enterCamTorum);
+		goTalkToJessamineAfterEyatlalli.addStep(inNeypotzli, talkToJessamineAfterEyatlalli);
+		goTalkToJessamineAfterEyatlalli.addStep(inCamTorum, enterNeypotzliAfterBlacksmith);
+		steps.put(19, goTalkToJessamineAfterEyatlalli);
+		steps.put(20, goTalkToJessamineAfterEyatlalli);
+		steps.put(21, goTalkToJessamineAfterEyatlalli);
+		steps.put(22, goTalkToJessamineAfterEyatlalli);
+
+		ConditionalStep getItemsWhenInStream = new ConditionalStep(this, getHerbloreSupplies);
+		getItemsWhenInStream.addStep(and(hadPaste, hadScales, rope), enterSmallEntrance);
+		getItemsWhenInStream.addStep(and(hadPaste, hadScales), getHunterSupplies);
+		getItemsWhenInStream.addStep(and(hadPaste, bream, knife), useKnifeOnBream);
+		getItemsWhenInStream.addStep(and(hadPaste, bigFishingNet), fishBream);
```

Comment:
Simplify this, and instead suggest where the user can get the knife from in case they've dropped it (next suggestion fixes this). Otherwise, the user can get stuck on the "fishBream" step
```suggestion
		getItemsWhenInStream.addStep(and(hadPaste, bream), useKnifeOnBream);
		getItemsWhenInStream.addStep(and(hadPaste, bigFishingNet), fishBream);
```